domains/certificates/Query.API/API/DataSyncSyncer/DataSyncService.cs

```diff
@@ -0,0 +1,123 @@
+
+    public DataSyncService(HttpClient httpClient, ILogger<DataSyncService> logger)
+    {
+        this.httpClient = httpClient;
+        this.logger = logger;
+    }
+
+    public void SetState(Dictionary<string, DateTimeOffset> state)
+    {
+        periodStartTimeDictionary = state;
+    }
+
+    public async Task<List<DataSyncDto>> FetchMeasurements(string GSRN, string meteringPointOwner,
+        CancellationToken cancellationToken)
+    {
+        var dateFrom = periodStartTimeDictionary[GSRN].ToUnixTimeSeconds();
+
+        var now = DateTimeOffset.UtcNow;
+        var midnight = new DateTimeOffset(now.Year, now.Month, now.Day, 0, 0, 0, TimeSpan.Zero).ToUnixTimeSeconds();
+        var result = new List<DataSyncDto>();
+
+        if (dateFrom < midnight)
+        {
+            try
+            {
+                result = await RequestDatasync(
+                    GSRN,
+                    new Period(
+                        DateFrom: dateFrom,
+                        DateTo: midnight
+                    ),
+                    meteringPointOwner,
+                    cancellationToken
+                );
+            }
+            catch (Exception e)
+            {
+                logger.LogInformation("An error occured: {error}, no measurements was fetched", e.Message);
+            }
+
+        }
+
+        SetNextPeriodStartTime(result, GSRN);
+        return result;
+    }
+
+    private void SetNextPeriodStartTime(List<DataSyncDto> measurements, string GSRN)
+    {
+        if (measurements.IsEmpty())
+        {
+            return;
+        }
+
+        var newestMeasurement = measurements.Max(m => m.DateTo);
+        periodStartTimeDictionary![GSRN] = DateTimeOffset.FromUnixTimeSeconds(newestMeasurement);
+    }
+
+    private async Task<List<DataSyncDto>> RequestDatasync(string gsrn, Period period, string meteringPointOwner,
+        CancellationToken cancellationToken)
+    {
+        logger.LogInformation(
+            "Fetching data in period from {from} to: {to}", period.DateFrom.ToString("MM/dd/yy H:mm:ss"), period.DateTo.ToString("MM/dd/yy H:mm:ss")
+        );
+
+        var url = $"measurements?gsrn={gsrn}&dateFrom={period.DateFrom}&dateTo={period.DateTo}";
+        httpClient.DefaultRequestHeaders.Authorization =
+            new AuthenticationHeaderValue("Bearer", GenerateToken(meteringPointOwner));
+
+        var response = await httpClient.GetAsync(url, cancellationToken);
+
+        return await response.Content.ReadFromJsonAsync<List<DataSyncDto>>(cancellationToken: cancellationToken)
+               ?? throw new Exception($"Fetch of measurements failed, base: {httpClient.BaseAddress} url: {url}");
+    }
+
+    private static string GenerateToken(string meteringPointOwner)
+    {
+        var claims = new Claim[]
+        {
+            new(JwtRegisteredClaimNames.UniqueName, "username"),
+            new(JwtRegisteredClaimNames.NameId, Guid.NewGuid().ToString()),
```

Comment:
This is the payload for a JWT from the auth service:

```
{
  "issued": "2022-11-17T06:46:29.147835+00:00",
  "expires": "2022-11-18T06:46:29.147835+00:00",
  "actor": "7f497b4a-8bfd-4ed4-a12f-6ab9235d998b",
  "subject": "c2410975-fa5f-441d-a40e-6231ed540495",
  "scope": [
    "meteringpoints.read",
    "measurements.read"
  ]
}
```

I think we should not add more claims than what is currently in the JWT token from the auth service, so I think we should remove these two claims

Reply:
and then add "issued" and "expires" claims. Lifetime validation is not performed by DataSync, but then we are doing the same as AuthService is currently doing